NebulaNetwork/Server.cs

```diff
@@ -66,6 +69,27 @@ public override void Start()
             PacketProcessor.SimulateLatency = true;
 #endif
 
+            if (Config.Options.EnableUPnpOrPmpSupport)
+            {
+                Task.Run(async () => {
+                    var discoverer = new NatDiscoverer();
+                    try
+                    {
+                        var device = await discoverer.DiscoverDeviceAsync();
```

Comment:
Hmm... Do you think it is a good idea to put the discover device part somewhere else (e.g. when the option is enabled or when preload completes) so it would take less time to start up server?

Reply:
I don't have the device so I don't know how long the time it takes. If it normally is really fast then you can ignore this suggestion.